openeo/extra/spectral_indices/spectral_indices.py

```diff
@@ -0,0 +1,99 @@
+from openeo.processes import ProcessBuilder, array_modify
+from openeo.rest.datacube import DataCube
+import numpy as np
+
+def _get_expression_map(cube: DataCube, x: ProcessBuilder):
+    collection_id = cube.graph["loadcollection1"]["arguments"]["id"]
```

Comment:
`loadcollection1` is not a very robust way to get collection id. It can easily be `loadcollection2` or something else.

I think it's more reliable to look into the datacube's metadata attribyte